Rename webrtc::pixelBufferFrom* functions to have 'create' in the name and fix a CVPixelBufferRef leak

smfr opened the pull request:
#### 11a2ba1a7a1c692d30a9ec41eb068c0f5bd76ffa
<pre>
Rename webrtc::pixelBufferFrom* functions to have &apos;create&apos; in the name and fix a CVPixelBufferRef leak
<a href="https://bugs.webkit.org/show_bug.cgi?id=263578">https://bugs.webkit.org/show_bug.cgi?id=263578</a>
rdar://117388614

Reviewed by Youenn Fablet.

Convention is that functions that return retained objects should have &apos;create&apos; or &apos;copy&apos; in the name.
Rename pixelBufferFromFrame(), pixelBufferFromI420Buffer() and pixelBufferFromI420ABuffer() to follow this, to allow for
style checker improvements in future.

This made it clear that there was a leak in RealtimeIncomingVideoSourceCocoa::toVideoFrame(), where
the retained result of webrtc::pixelBufferFromFrame() was passed to createVideoSampleFromCVPixelBuffer(),
leaking the CVPixelBufferRef.

* Source/ThirdParty/libwebrtc/Configurations/libwebrtc.exp:
* Source/ThirdParty/libwebrtc/Source/webrtc/sdk/WebKit/WebKitUtilities.h:
* Source/ThirdParty/libwebrtc/Source/webrtc/sdk/WebKit/WebKitUtilities.mm:
(webrtc::createPixelBufferFromI420Buffer):
(webrtc::createPixelBufferFromI420ABuffer):
(webrtc::copyPixelBufferForFrame):
(webrtc::pixelBufferFromI420Buffer): Deleted.
(webrtc::pixelBufferFromI420ABuffer): Deleted.
(webrtc::pixelBufferFromFrame): Deleted.
* Source/WebCore/platform/graphics/cv/VideoFrameCV.mm:
(WebCore::VideoFrame::createI420):
(WebCore::VideoFrame::createI420A):
* Source/WebCore/platform/mediastream/mac/RealtimeIncomingVideoSourceCocoa.h:
* Source/WebCore/platform/mediastream/mac/RealtimeIncomingVideoSourceCocoa.mm:
(WebCore::RealtimeIncomingVideoSourceCocoa::createVideoSampleFromCVPixelBuffer):
(WebCore::RealtimeIncomingVideoSourceCocoa::toVideoFrame):
* Source/WebKit/WebProcess/GPU/webrtc/SharedVideoFrame.cpp:
(WebKit::SharedVideoFrameWriter::writeBuffer):

Canonical link: <a href="https://commits.webkit.org/269747@main">https://commits.webkit.org/269747@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c538b090b3017c9d962b3b3378a074f70a5a0834

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/23458 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/1572 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/24582 "Built successfully") | [⏳ 🛠 wpe ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/21658 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/23727 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/3119 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/24000 "Built successfully") | [⏳ 🧪 wpe-wk2 ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/23700 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/1122 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/20302 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/26216 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/924 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/21201 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/27580 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/21381 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/21459 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/25237 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/909 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/18630 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/872 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5597 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/1319 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/1179 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/ThirdParty/libwebrtc/Source/webrtc/sdk/WebKit/WebKitUtilities.mm:
```diff
@@ -317,7 +317,7 @@ CVPixelBufferRef createPixelBufferFromFrameBuffer(VideoFrameBuffer& buffer, cons
 }
 
-CVPixelBufferRef pixelBufferFromFrame(const VideoFrame& frame)
+CVPixelBufferRef createPixelBufferFromFrame(const VideoFrame& frame)
```
We are not really creating the pixel buffer here.
How about keeping `pixelBufferFromFrame` name, removing the CVPixelBufferRetain call below, and updating the call sites accordingly.

smfr replied:
The "create" indicates this returns a retained buffer. If not create, then it needs "copy" or "retain" in the name.

smfr replied:
I named it `copyPixelBufferForFrame()`

smfr committed:
Rename webrtc::pixelBufferFrom* functions to have 'create' in the name and fix a CVPixelBufferRef leak
https://bugs.webkit.org/show_bug.cgi?id=263578
rdar://117388614

Reviewed by Youenn Fablet.

Convention is that functions that return retained objects should have 'create' or 'copy' in the name.
Rename pixelBufferFromFrame(), pixelBufferFromI420Buffer() and pixelBufferFromI420ABuffer() to follow this, to allow for
style checker improvements in future.

This made it clear that there was a leak in RealtimeIncomingVideoSourceCocoa::toVideoFrame(), where
the retained result of webrtc::pixelBufferFromFrame() was passed to createVideoSampleFromCVPixelBuffer(),
leaking the CVPixelBufferRef.

* Source/ThirdParty/libwebrtc/Configurations/libwebrtc.exp:
* Source/ThirdParty/libwebrtc/Source/webrtc/sdk/WebKit/WebKitUtilities.h:
* Source/ThirdParty/libwebrtc/Source/webrtc/sdk/WebKit/WebKitUtilities.mm:
(webrtc::createPixelBufferFromI420Buffer):
(webrtc::createPixelBufferFromI420ABuffer):
(webrtc::copyPixelBufferForFrame):
(webrtc::pixelBufferFromI420Buffer): Deleted.
(webrtc::pixelBufferFromI420ABuffer): Deleted.
(webrtc::pixelBufferFromFrame): Deleted.
* Source/WebCore/platform/graphics/cv/VideoFrameCV.mm:
(WebCore::VideoFrame::createI420):
(WebCore::VideoFrame::createI420A):
* Source/WebCore/platform/mediastream/mac/RealtimeIncomingVideoSourceCocoa.h:
* Source/WebCore/platform/mediastream/mac/RealtimeIncomingVideoSourceCocoa.mm:
(WebCore::RealtimeIncomingVideoSourceCocoa::createVideoSampleFromCVPixelBuffer):
(WebCore::RealtimeIncomingVideoSourceCocoa::toVideoFrame):
* Source/WebKit/WebProcess/GPU/webrtc/SharedVideoFrame.cpp:
(WebKit::SharedVideoFrameWriter::writeBuffer):

Canonical link: https://commits.webkit.org/269747@main
